src/services/passwordGeneration.service.ts

```diff
@@ -217,7 +223,82 @@ export class PasswordGenerationService implements PasswordGenerationServiceAbstr
             }
         }
 
-        return this.optionsCache;
+        const enforcedPolicyOptions = await this.getPasswordGeneratorPolicyOptions();
+        if (this.optionsCache.length < enforcedPolicyOptions.minLength) {
+            this.optionsCache.length = enforcedPolicyOptions.minLength;
+        }
+        if (enforcedPolicyOptions.useUppercase) {
+            this.optionsCache.uppercase = true;
+        }
+        if (enforcedPolicyOptions.useLowercase) {
+            this.optionsCache.lowercase = true;
+        }
+        if (enforcedPolicyOptions.useNumbers) {
+            this.optionsCache.number = true;
+        }
+        if (this.optionsCache.minNumber < enforcedPolicyOptions.numberCount) {
+            this.optionsCache.minNumber = enforcedPolicyOptions.numberCount;
+        }
+        if (enforcedPolicyOptions.useSpecial) {
+            this.optionsCache.special = true;
+        }
+        if (this.optionsCache.minSpecial < enforcedPolicyOptions.specialCount) {
+            this.optionsCache.minSpecial = enforcedPolicyOptions.specialCount;
+        }
+        // Must normalize these fields because the receiving call expects all options to pass the current rules
+        if (this.optionsCache.minSpecial + this.optionsCache.minNumber > this.optionsCache.length) {
+            this.optionsCache.minSpecial = this.optionsCache.length - this.optionsCache.minNumber;
+        }
+
+        return [this.optionsCache, enforcedPolicyOptions];
+    }
+
+    async getPasswordGeneratorPolicyOptions(): Promise<PasswordGeneratorPolicyOptions> {
+        const policies: Policy[] = await this.policyService.getAll(PolicyType.PasswordGenerator);
+        const enforcedOptions: PasswordGeneratorPolicyOptions = new PasswordGeneratorPolicyOptions();
+
+        if (policies == null || policies.length === 0) {
+            return enforcedOptions;
+        }
+
+        policies.forEach((currentPolicy) => {
+            if (!currentPolicy.enabled || currentPolicy.data == null) {
+                return;
+            }
+            const currentPolicyData = currentPolicy.data;
```

Comment:
Not really necessary. Saving 1 character.

Reply:
Fixed in next update